Mage.Sets/src/mage/cards/c/CaptainsManeuver.java

```diff
@@ -56,7 +56,7 @@ class CaptainsManeuverEffect extends RedirectionEffect {
     protected MageObjectReference redirectToObject;
 
     public CaptainsManeuverEffect() {
-        super(Duration.EndOfTurn, Integer.MAX_VALUE, UsageType.ONE_USAGE_ABSOLUTE);
+        super(Duration.EndOfTurn, Integer.MAX_VALUE, UsageType.ACCORDING_DURATION);
```

Comment:
Can we standardize how we should handle RedirectionEffects with variable amount of redirection? I see some have `0` while this one has `Integer.MAX_VALUE`. In reality, neither are used.

Maybe we should have a new class that extends RedirectionEffect, VariableRedirectionEffect, which provides API for setting redirectionAmount (an overridable abstract function perhaps). This could be a separate PR

Reply:
I used some X-cost spell as a reference when making this one, which had this temp value set to MAX_VALUE, but don't remember what it was anymore.

In any case, redirection might not be something that necessarily needs new features, because it's so rare for new redirection cards to come out. Most of them are from the 90's and 00's.

Reply:
I think that kind of refactoring should be its own PR.